components/sst_importer/src/sst_importer.rs

```diff
@@ -535,18 +533,24 @@ impl SstImporter {
         backend: &StorageBackend,
         speed_limiter: &Limiter,
     ) -> Result<Arc<Vec<u8>>> {
-        if self.import_support_download() {
-            let tmp_file = self.do_download_kv_file(meta, backend, speed_limiter)?;
-            let file = File::open(tmp_file)?;
-            let mut reader = BufReader::new(file);
-            let mut buffer = Vec::new();
-            reader.read_to_end(&mut buffer)?;
-
-            let rewrite_buff = self.rewrite_kv_file(buffer, rewrite_rule)?;
-            Ok(Arc::new(rewrite_buff))
+        let c = if self.import_support_download() {
+            self.do_download_kv_file(meta, backend, speed_limiter)?
         } else {
-            let buffer = self.do_read_kv_file(meta, rewrite_rule, ext_storage, speed_limiter)?;
-            Ok(buffer)
+            self.do_read_kv_file(meta, rewrite_rule, ext_storage, speed_limiter)?
+        };
+        match c {
+            // If cache memroy, it has been rewrite, return buffer directly.
+            CacheKvFile::Mem(buff) => Ok(buff),
+            // If cache file name, it need to read and rewrite.
+            CacheKvFile::Fs(path) => {
+                let file = File::open(path.as_ref())?;
+                let mut reader = BufReader::new(file);
+                let mut buffer = Vec::new();
+                reader.read_to_end(&mut buffer)?;
+
+                let rewrite_buff = self.rewrite_kv_file(buffer, rewrite_rule)?;
+                Ok(Arc::new(rewrite_buff))
+            }
```

Comment:
Perhaps we can make this a method of `CacheKvFile`, like `CacheKvFile::get_content(&self)`?